vspec/model/constants.py

```diff
@@ -10,127 +10,159 @@
 #
 # Constant Types and Mappings
 #
-
 # noinspection PyPackageRequirements
-from enum import Enum, EnumMeta
-from typing import Sequence, Type, TypeVar
+import re
+import pkg_resources
+from typing import (
+    Sequence, Type, TypeVar, Optional, Dict, Tuple, NamedTuple, Iterator,
+)
+
+import yaml
+
+from vspec.model import CONFIG_PATH, NON_ALPHANUMERIC_WORD
 
 
 T = TypeVar("T")
 
 
-class EnumMetaWithReverseLookup(EnumMeta):
-    """This class extends EnumMeta and adds:
+class VSSConstant(NamedTuple):
+    label: str
+    value: str
+    description: Optional[str] = None
+    domain: Optional[str] = None
+
+
+def dict_to_constant_config(name: str, info: Dict[str, str]) -> Tuple[str, VSSConstant]:
+    label = info['label']
+    label = NON_ALPHANUMERIC_WORD.sub('', label).upper()
+    description = info.get('description', None)
+    domain = info.get('domain', None)
+    return label, VSSConstant(info['label'], name, description, domain)
+
+
+def iterate_config_members(config: Dict[str, Dict[str, str]]) -> Iterator[Tuple[str, VSSConstant]]:
+    for u, v in config.items():
+        yield dict_to_constant_config(u, v)
+
+
+class VSSEnumMeta(type):
+    """This class defines the enumeration behavior for vss:
+     - Access through Class.ATTRIBUTE
      - from_str(str): reverse lookup
      - values(): sequence of values
     """
-    def __new__(typ, *args, **kwargs):
-        cls = super().__new__(typ, *args, **kwargs)
-        if not hasattr(cls, "__reverse_lookup__"):
+
+    def __new__(mcs, cls, bases, classdict):
+        cls = super().__new__(mcs, cls, bases, classdict)
+
+        if not hasattr(cls, '__reverse_lookup__'):
             cls.__reverse_lookup__ = {
-                v.value: v for v in cls.__members__.values()
+                v.value: v for k, v in cls.__config__.items()
             }
-        if not hasattr(cls, "__values__"):
-            cls.__values__ = tuple(v.value for v in cls.__members__.values())
+        if not hasattr(cls, '__values__'):
+            cls.__values__ = set(cls.__reverse_lookup__.keys())
```

Comment:
it should be `list()` as the iteration order will remain constant (with set it will only be until more items are added). Unfortunately to add new entries you must check if they exist before and then add to the end (or keep it sorted?)

Reply:
also, `add_config()` should be in this meta class

Reply:
We have now `__values__` as a list